pkg/v1/providers/infrastructure-azure/v1.2.1/ytt/overlay.yaml

```diff
@@ -36,6 +38,14 @@
 #@ return data.values.AZURE_IDENTITY_NAMESPACE
 #@ end
 
+#@ def verify_and_configure_strategy(strategytype):
+#@   if strategytype not in strateg_type:
+#@     assert.fail("Invalid Strategy type  \""+strategytype+"\", it must be OnDelete or RollingUpdate")
+#@   end
+#@   return strategytype
+#@ end
+
+
```

Comment:
Dedupe please, please put this into common library in pkg/v1/providers/lib/helpers.star as

```starlark
valid_strategy_types = ["OnDelete", "RollingUpdate"]


def verify_and_configure_machine_deployment_rollout_strategy(strategy_type):
if strategy_type not in valid_strategy_types:
     assert.fail("Invalid Strategy type  \""+strategytype+"\", it must be OnDelete or RollingUpdate")
   end
   return strategy_type
end
```

and remove from all overlays